src/main/java/com/autotune/database/helper/DBHelpers.java

```diff
@@ -217,27 +217,40 @@ public static void setRecommendationsToKruizeObject(List<ListRecommendationsAPIO
                 for (K8sObject k8sObject : kruizeObject.getKubernetes_objects()) {
                     if (null == kubernetesAPIObject.getName()) {
                         // namespace recommendations experiment type
-                        NamespaceData namespaceData = k8sObject.getNamespaceData();
+                        if (kubernetesAPIObject.getName().equalsIgnoreCase(k8sObject.getName())
```

Comment:
This check here seems incorect. It will give NPE.
Check if it's really required.

Reply:
right, removed this faulty check